ts/components/services/ServicePreferenceSummary.tsx

```diff
@@ -10,27 +10,45 @@ import { Link } from "../core/typography/Link";
 import { GlobalState } from "../../store/reducers/types";
 import { Dispatch } from "../../store/actions/types";
 import { navigateToServicePreferenceScreen } from "../../store/actions/navigation";
+import { ServicesPreferencesModeEnum } from "../../../definitions/backend/ServicesPreferencesMode";
+import { profileServicePreferencesModeSelector } from "../../store/reducers/profile";
 
 type Props = ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps>;
 
-const mapChoiceLabel = new Map<string, string>([
-  ["AUTO", I18n.t("services.optIn.preferences.quickConfig.value")],
-  ["MANUAL", I18n.t("services.optIn.preferences.manualConfig.value")]
-]);
-
-const getChoiceLabel = (status: string) =>
-  mapChoiceLabel.get(status) ??
-  I18n.t("services.optIn.preferences.unavailable");
+export const mapServicesPreferenceModeLabel: Record<
+  ServicesPreferencesModeEnum,
+  string
+> = {
+  [ServicesPreferencesModeEnum.AUTO]: I18n.t(
+    "services.optIn.preferences.quickConfig.value"
+  ),
+  [ServicesPreferencesModeEnum.MANUAL]: I18n.t(
+    "services.optIn.preferences.manualConfig.value"
+  ),
+  [ServicesPreferencesModeEnum.LEGACY]: I18n.t(
+    "services.optIn.preferences.unavailable"
+  )
+};
 
+/**
+ * Display the current services preference mode set in the profile
+ * Display a link to navigate to a screen to update the service preference mode
+ * @param props
+ * @constructor
+ */
 const ServicePreferenceSummary = (props: Props): React.ReactElement => (
-  // FIXME Replace the string with the current preference option saved on profile
   <View style={IOStyles.horizontalContentPadding}>
     <View spacer />
     <InfoBox iconName={"io-info"} iconColor={IOColors.bluegrey} iconSize={24}>
       <H5 weight={"Regular"} color={"bluegrey"}>{`${I18n.t(
         "services.optIn.preferences.choiceLabel"
-      )} ${getChoiceLabel(props.serviceOption)}`}</H5>
+      )} ${
+        mapServicesPreferenceModeLabel[
```

Comment:
To prevent issues with translations this value should be returned by a function this is what's happening:
Telephone lang settings: Italian
App settings: English

<img src="https://user-images.githubusercontent.com/3959405/124307480-2ba13900-db68-11eb-937b-9c7e407bd7a1.png" height="550"/>

Reply:
I totally forget this pain point.
Good catch
changed in https://github.com/pagopa/io-app/pull/3191/commits/60f368a6457d76ea601d89aa0454910ebd07b5e1